File: order-producer/src/main/java/com/sahakav/coffee_shop/model/CoffeeOrder.java
package com.sahakav.coffee_shop.model;

import javax.persistence.*;
import java.util.UUID;

@Entity
@Table(name = "coffee_order")
public class CoffeeOrder {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;

    @Column(name = "type")
    @Enumerated(EnumType.STRING)
    private CoffeeType coffeeType;

    public CoffeeOrder(CoffeeType coffeeType) {
        this.coffeeType = coffeeType;
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public CoffeeType getCoffeeType() {
        return coffeeType;
    }

    public void setCoffeeType(CoffeeType coffeeType) {
        this.coffeeType = coffeeType;
    }
}

File: order-producer/src/main/scala/com/sahakav/coffee_shop/model/CoffeeOrder.scala
package com.sahakav.coffee_shop.model

import javax.persistence._
import java.util.UUID


@Entity
@Table(name = "coffee_order")
class CoffeeOrder {
  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  var id: UUID = _

  @Enumerated(EnumType.STRING)
  @Column(name = "orderState")
  var orderState: CoffeeOrderState = CoffeeOrderState.IN_PROCESS

  @Column(name = "coffeeType")
  var coffeeType: CoffeeType = _


}

object CoffeeOrder {

  def apply(coffeeType: CoffeeType): CoffeeOrder = {
    val coffeeOrder: CoffeeOrder = new CoffeeOrder
    coffeeOrder.coffeeType = coffeeType
    coffeeOrder
  }

}

